src/status_im2/contexts/onboarding/profiles/view.cljs

```diff
@@ -1,7 +1,203 @@
+       :key-fn                  :key-uid
+       :content-container-style {:padding-bottom 20}
+       :render-fn               profile-card}]]))
+
+(defn login-section
+  []
+  (let [{:keys [name customization-color error processing]
+         :as   multiaccount} (rf/sub [:multiaccounts/login])
+        sign-in-enabled?     (rf/sub [:sign-in-enabled?])
+        profile-picture      (:uri (first (:images multiaccount)))]
+    [rn/keyboard-avoiding-view
+     {:style style/login-container}
+     [rn/view
+      {:style {:flex 1}}
+      [quo/button
+       {:size                32
+        :type                :blur-bg
+        :icon                true
+        :on-press            #(reset! show-profiles? true)
+        :override-theme      :dark
+        :width               32
+        :accessibility-label :show-profiles
+        :style               style/multi-profile-button}
+       :i/multi-profile]
+      [quo/profile-card
+       {:name                name
+        :customization-color (or customization-color :primary)
+        :profile-picture     (when profile-picture {:uri profile-picture})
+        :card-style          style/login-profile-card}]
+      [quo/input
+       {:type              :password
+        :blur?             true
+        :override-theme    :dark
+        :placeholder       (i18n/label :t/type-your-password)
+        :auto-focus        true
+        :error?            (when (not-empty error) error)
+        :label             (i18n/label :t/profile-password)
+        :secure-text-entry true
+        :on-change-text    (fn [password]
+                             (rf/dispatch [:set-in [:multiaccounts/login :password]
+                                           (security/mask-data password)])
+                             (rf/dispatch [:set-in [:multiaccounts/login :error] ""]))
+        :on-submit-editing (when sign-in-enabled? login-multiaccount)}]
+      (when (not-empty error)
+        [quo/info-message
+         {:type  :error
+          :size  :default
+          :icon  :i/info
+          :style style/info-message}
+         (i18n/label :t/oops-wrong-password)])]
+     [quo/button
+      {:size                40
+       :type                :ghost
+       :before              :i/info
+       :accessibility-label :forget-password-button
+       :override-theme      :dark}
+      (i18n/label :t/forget-password)]
+     [quo/button
+      {:size                40
+       :type                :primary
+       :customization-color (or :primary customization-color)
+       :accessibility-label :login-button
+       :override-theme      :dark
+       :before              :i/unlocked
+       :disabled            (or (not sign-in-enabled?) processing)
+       :on-press            login-multiaccount
+       :style               (style/login-button)}
+      (i18n/label :t/log-in)]]))
 
 (defn views
   []
-  [:<>
-   [background/view true]])
+  (reset! show-profiles? false)
```

Comment:
Hi @ulisesmac,
> pass the derefed value as usually

If we pass the value as derefed instead of passing the atom itself, wouldn't we be unable to reset it.
Or am I misunderstanding what you meant?

> IMO it's a cleaner pattern to declare local atoms.

Both sections' profile & login needs access to this atom. And for the profiles section, we need to pass it to every item of the flat list so they can reset it. Maybe this is only case specific, but having it global looked cleaner.

> but if we would have more atoms, and all of them were global, it'd be hard to track them across the views

Please can you elaborate on what you meant here?

btw, just curious about the reasoning, reagent atom is just temporary we are going to remove it. ([comment](https://github.com/status-im/status-mobile/pull/15376#discussion_r1141798943))

Reply:
Hi @Parveshdhull 

> If we pass the value as derefed instead of passing the atom itself, wouldn't we be unable to reset it.
Or am I misunderstanding what you meant?

> Both sections' profile & login needs access to this atom. And for the profiles section, we need to pass it to every item of the flat list so they can reset it. Maybe this is only case specific, but isn't having it global looks cleaner here? (also easy to remove)

Yep, you're  correct, we need to update the atom's value, but it is usually done by using a callback function (very common in React components).

For this component, it could look as follows:
```clojure
(defn views
  []
  (let [show-profiles? (reagent/atom false)] ;;<- atom initialized
    (fn []
      [:<>
       [background/view true]
       (if @show-profiles?
         [profiles-section {:on-card-press (fn the-callback [])}] ;; <- function to handle the press event
         [login-section {:on-press #(reset! show-profiles? true)}] ;; <- function to handle the press event
         )])))
```
So we receive that callback as a parameter in those components.

>Please can you elaborate on what you meant here?

Yes! I meant that working with global vars that are used across multiple functions leads us to a more complex code. Instead using local bindings (as much as we can) usually makes code simpler (and as a plus, we are a little closer to writing pure functions).

> btw, just curious about the reasoning, reagent atom is just temporary we are going to remove it.

Thanks for the context! :+1:  good to know!


As a side clean up note, the code currently is receiving a ratom but is not used, it's looking as:
```clojure
(if @show-profiles?
       [profiles-section]
       [login-section show-profiles?]) ;; <- ratom passed to this component
       
;; but:

(defn login-section
  [] ;; <- vector is empty
  (let [ ]
    ))
```

Reply:
sorry, will remove it.
(leftover of approach when I was trying to use local atom instead of global one)